### PR TITLE
Helm docs 1026

### LINKED ISSUE #1026: SPK service create helm chart needs better documentation
**As a:**
Developer
**I want:**
To understand the `spk service create` helm chart flags
**So that:**
I can stop thinking about how to configure `spk` and get on with my business problem 
**Describe the solution you'd like:**
Currently the help text says
```
-c, --helm-chart-chart <helm-chart>
bedrock helm chart name. --helm-chart-* and --helm-config-* are exclusive; you may only use one.
-r, --helm-chart-repository <helm-repository>
bedrock helm chart repository. --helm-chart-* and --helm-config-* are exclusive; you may only use one.
-b, --helm-config-branch <helm-branch>
bedrock custom helm chart configuration branch. --helm-chart-* and --helm-config-* are exclusive; you may only use one.
-p, --helm-config-path <helm-path>
bedrock custom helm chart configuration path. --helm-chart-* and --helm-config-* are exclusive; you may only use one.
-g, --helm-config-git <helm-git>
bedrock helm chart configuration git repository. --helm-chart-* and --helm-config-* are exclusive; you may only use one.
```

This is confusing. The [spk project guide](https://github.com/CatalystCode/spk/blob/master/guides/project-service-management-guide.md#adding-a-service-to-a-application-repository) doesn't mention how to use these flags  It would be great to see sample scenarios for how to use these __mutually exclusive__ flags. 

_**Consider**_
The following to indicate mutually exclusivity:
```{--helm-chart-chart chart-name --helm-chart-repository repo-url | --helm-config-branch branch --helm-config-path path --helm-config-git helm-git}```

Above there are two options:
Left side of pipe: Explain why and when you would use `--helm-chart-chart` and `--helm-chart-repository`
Right side of pipe: Explain why and when you would use `--helm-config-branch`, `--helm-config-path path` and `--helm-config-git`

**Acceptance Criteria:**

- Sample usage for external helm charts and internal (local to the application repo) helm chart scenarios in the [docs](https://catalystcode.github.io/spk/commands/index.html#service_create).
- Guidance in the [spk project guide](https://github.com/CatalystCode/spk/blob/master/guides/project-service-management-guide.md#adding-a-service-to-a-application-repository)  on how to use these flags
- Consider not having mutually exclusive flags. I still don't understand why `spk service create` has this complication. If absolutely necessary we should explain.

**Does this require updates to documentation?:**

yes

### PR DESCRIPTION
**Pull request by bnookala**
Fixes https://github.com/microsoft/bedrock/issues/1026

### CODE REVIEW

**Review by yradsmikham — APPROVED** [review]

File: guides/project-service-management-guide.md [hunk 1]
```diff
@@ -20,8 +20,8 @@ This document describes the workflow for deploying a set of services
   configured to sync from the single Materialized Manifest Repositories, and
   multiple Project repositories can be pointed to the single High Level
   Definition Repository.
-- Step 5 can be repeated each time you need to onboard a service to your Bedrock
-  automated infrastructure.
+- Step 5 can be repeated each time you need to onboard a service to your
+  Bedrxock automated infrastructure.
```
Comment (mtarng) [inline guides/project-service-management-guide.md]: typo
> Bedrxock

File: guides/project-service-management-guide.md [hunk 2]
```diff
@@ -251,6 +251,72 @@ application repositories
 **NOTE** `spk service` command documentation can be found
 [here](/guides/service-management.md).
 
+#### Helm Configuration for SPK
+
+`spk service create` allows a user to configure a service a number fo ways with
```
Comment (mtarng) [inline guides/project-service-management-guide.md]: typo
> fo

File: guides/project-service-management-guide.md [hunk 3]
```diff
@@ -251,6 +251,72 @@ application repositories
...
+
+As an `spk` user, if you would like to incorporate helm charts from a well known
+public repository, you may simply run `spk` the following `helm-chart`
+arguments:
+
+```
+spk service create --helm-chart-chart stable/nginx --helm-chart-repository github.com/helm/charts
+```
+
+##### Helm Charts in a distinct Git Repository
```
Comment (mtarng) [inline guides/project-service-management-guide.md]: Do we want to include a blurb here about PATs and access?

File: guides/project-service-management-guide.md [hunk 4]
```diff
@@ -251,6 +251,72 @@ application repositories
...
+source (no distinct git repository), you may still use the `helm-config`
+arguments to configure `spk`.
+
+If you presume that the `fabrikam-app` repository is the _same_ repository as
+the repository of your application sources, then the following invocation will
+allow a user to configure a service with helm charts tracked in the _same_
+repository.
+
+```
+spk service create --helm-config-git https://dev.azure.com/fabrikam/fabrikam-project/_git/fabrikam-app --helm-config-branch master --helm-path /charts/fabrikam
```
Comment (mtarng) [inline guides/project-service-management-guide.md]: Do we want to include a blurb here and above that these git repo links cannot have a user in them?

**Review by mtarng — DISMISSED** [review]

**Review by evanlouie — APPROVED** [review]

File: guides/project-service-management-guide.md (hunk 3 above)
Comment (andrebriggs) [reply guides/project-service-management-guide.md]: Agree with @mtarng

Comment (bnookala) [reply guides/project-service-management-guide.md]: fixed

File: guides/project-service-management-guide.md (hunk 4 above)
Comment (bnookala) [reply guides/project-service-management-guide.md]: fixed

File: guides/project-service-management-guide.md [hunk 5]
```diff
@@ -251,6 +251,136 @@ application repositories
...
+parameters to `spk service create --helm-config-git`:
+
+`https://dev.azure.com/fabrikam/fabrikam-project/_git/fabrikam-helm-charts`
+
+##### Helm Charts in a distinct Git Repository from Application Sources in a different Azure DevOps Project
+
+If your Helm Charts are in their own distinct Git Repository in a different
+Azure DevOps project, you can use still use the `helm-config` arguments to
+configure `spk`, but must also provide another option,
+`--helm-config-access-token-variable`. This configuration option is the name of
```
Comment (evanlouie) [inline guides/project-service-management-guide.md:338]: Might want to reference how in the happy-path workflow, the default value that gets injected of `ACCESS_TOKEN_SECRET` will work be default for all projects in their org (assuming they've given that permission to their token)

File: guides/project-service-management-guide.md [hunk 6]
```diff
@@ -251,6 +251,136 @@ application repositories
...
+the environment variable containing the Personal Access Token to access the git
+repository in `helm-config-git`:
+
+```
+spk service create --helm-config-git https://dev.azure.com/fabrikam/fabrikam-helm-charts-project/_git/fabrikam-helm-charts --helm-config-branch master --helm-path /charts/fabrikam --helm-config-access-token-variable FABRIKAM_HELM_CHARTS_REPO_PAT
```
Comment (evanlouie) [inline guides/project-service-management-guide.md]: might want to change to:

```
spk service create \
  --helm-config-git https://dev.azure.com/fabrikam/fabrikam-helm-charts-project/_git/fabrikam-helm-charts \
  --helm-config-branch master \
  --helm-path /charts/fabrikam \
  --helm-config-access-token-variable FABRIKAM_HELM_CHARTS_REPO_PAT
```

for legibility

File: guides/project-service-management-guide.md [hunk 7]
```diff
@@ -251,6 +251,136 @@ application repositories
...
+source (no distinct git repository), you may still use the `helm-config`
+arguments to configure `spk`.
+
+If you presume that the `fabrikam-app` repository is the _same_ repository as
+the repository of your application sources, then the following invocation will
+allow a user to configure a service with helm charts tracked in the _same_
+repository.
+
+```
+spk service create --helm-config-git https://dev.azure.com/fabrikam/fabrikam-project/_git/fabrikam-app --helm-config-branch master --helm-path /charts/fabrikam
```
Comment (evanlouie) [inline guides/project-service-management-guide.md]: maybe change to:
```
spk service create \
  --helm-config-git https://dev.azure.com/fabrikam/fabrikam-project/_git/fabrikam-app \
  --helm-config-branch master \
  --helm-path /charts/fabrikam
```
for legibility

File: guides/project-service-management-guide.md [hunk 8]
```diff
@@ -251,6 +251,136 @@ application repositories
...
+spk service create --helm-chart-chart stable/nginx --helm-chart-repository github.com/helm/charts
+```
+
+##### Helm Charts in a distinct Git Repository from Application Sources in the same Azure DevOps Project
+
+If your Helm Charts are in their own distinct Git Repository in the _same_ Azure
+DevOps project, you can use the `helm-config` arguments to configure `spk`:
+
+```
+spk service create --helm-config-git https://dev.azure.com/fabrikam/fabrikam-project/_git/fabrikam-helm-charts --helm-config-branch master --helm-path /charts/fabrikam
```
Comment (evanlouie) [inline guides/project-service-management-guide.md]:
```
spk service create \
  --helm-config-git https://dev.azure.com/fabrikam/fabrikam-project/_git/fabrikam-helm-charts \
  --helm-config-branch master \
  --helm-path /charts/fabrikam
```
for legibility

**Pull request merged**